### PR TITLE
(prometheus) prevent error to use $__interval_ms in query

### PR DESCRIPTION
**Pull request by mtanda**
`$__interval_ms` type is integer. It cause error.

### CODE REVIEW

**Comment by davkal** [comment]
This function could use a comment and a test. Currently I have no idea why it's there and what input/output is typical.

**Review by torkelo — COMMENTED** [review]
thanks @mtanda . A test for this and maybe a comment what this function does. Should this function not always return a string?

**Comment by mtanda** [comment]
Sorry, I'm not sure what I should do.

There are tests for `prometheusRegularEscape()`.
https://github.com/grafana/grafana/blob/v5.2.1/public/app/plugins/datasource/prometheus/specs/datasource.jest.ts#L168-L178

I think `prometheusRegularEscape()` should return string type.
So I change the code, convert number to string in caller side.

**Comment by davkal** [comment]
@mtanda we would like to use opportunities where an error was encountered to extend our unit tests.
In this instance it happened in `interpolateQueryExpr`. Could we ask you to please add a test suite for `interpolateQueryExpr()` inside `datasource.jest.ts` as part of this patch? Ideally it would contain a test that would fail before your fix (and pass with the fix).

**Comment by mtanda** [comment]
I understand what you need. I add a test case for failure pattern.

File: public/app/plugins/datasource/prometheus/datasource.ts [hunk 1]
```diff
@@ -103,6 +103,10 @@ export class PrometheusDatasource {
   interpolateQueryExpr(value, variable, defaultFormatFn) {
     // if no multi or include all do not regexEscape
     if (!variable.multi && !variable.includeAll) {
+      if (typeof value === 'number') {
+        // $__interval_ms type is number, convert to string here
+        value = value.toString();
```
Comment (torkelo) [inline public/app/plugins/datasource/prometheus/datasource.ts]: not sure this is the correct place to solve this, maybe the root issue should be fixed instead? wonder if allowing $__interval_ms to have number value is not correct but it should be turned into string when it is save in scopedVars (in metrics_ctrl & prometheus datasource.ts)

**Pull request merged**